connector/src/it/scala/com/datastax/spark/connector/cql/ContinuousPagingScannerSpec.scala

```diff
@@ -76,6 +83,40 @@ class ContinuousPagingScannerSpec extends SparkCassandraITFlatSpecBase with Defa
 
     withClue(sessions.map(_.toString).mkString("\n"))(sessions.size should be(1))
   }
+
+  private def executeContinuousPagingScan(readConf: ReadConf): Statement[_] = {
+    // we don't want to use the session from CC as mockito is unable to spy on a Proxy
+    val cqlSession = conn.conf.connectionFactory.createSession(conn.conf)
+    try {
+      val sessionSpy = spy(cqlSession)
+      val scanner = ContinuousPagingScanner(readConf, conn.conf, IndexedSeq.empty, sessionSpy)
+      val stmt = sessionSpy.prepare(s"SELECT * FROM $ks.test1").bind()
+      val statementCaptor = ArgumentCaptor.forClass(classOf[Statement[_]])
+
+      scanner.scan(stmt)
+      verify(sessionSpy).executeContinuously(statementCaptor.capture())
+      statementCaptor.getValue
+    } finally {
+      cqlSession.close()
+    }
+  }
+
+  it should "apply MB/s throughput limit" in skipIfNotDSE(conn) {
+    val executedStmt = executeContinuousPagingScan(ReadConf(throughputMiBPS = Some(32.0)))
+
+    executedStmt.getExecutionProfile.getBoolean(DseDriverOption.CONTINUOUS_PAGING_PAGE_SIZE_BYTES) should be(true)
+    executedStmt.getExecutionProfile.getInt(DseDriverOption.CONTINUOUS_PAGING_MAX_PAGES_PER_SECOND) should be (1000)
+    executedStmt.getExecutionProfile.getInt(DseDriverOption.CONTINUOUS_PAGING_PAGE_SIZE) should be(33554) // 32MB/s
+  }
+
+  it should "apply reads/s throughput limit" in skipIfNotDSE(conn) {
+    val executedStmt = executeContinuousPagingScan(
+      ReadConf(fetchSizeInRows = 999, readsPerSec = Some(5)))
+
+    executedStmt.getExecutionProfile.getBoolean(DseDriverOption.CONTINUOUS_PAGING_PAGE_SIZE_BYTES) should be(false)
+    executedStmt.getExecutionProfile.getInt(DseDriverOption.CONTINUOUS_PAGING_MAX_PAGES_PER_SECOND) should be(5)
+    executedStmt.getExecutionProfile.getInt(DseDriverOption.CONTINUOUS_PAGING_PAGE_SIZE) should be(999)
+  }
 }
```

Comment:
Perhaps add tests for failing cases?
e.g.
cases that may lead to `bytesPerPage <= 0`

Reply:
done